### ISSUE TITLE
Twiss of thick model with particle charge

### ISSUE THREAD

**Issue by RebTaylor** [xsuite/xsuite#369]
Performing Twiss of ions (i.e. particles with defined mass and charge) for the thin-lens model works, but for the thick-lens model of the same beam, get error 'ClosedOrbitSearchError:'.

Using the following particle information
```
mass = 11.177929 * 1E9 # GeV
pc = 0.951303*12 * 1E9 # GeV
charge = 6.0
particles = xp.Particles(mass0=mass, q0=charge, p0c=pc, nparticles=100)
```
Twiss for these variables works with MADX and PTC.
Using mass_ratio and charge_ratio, instead of mass0 and q0, gives the same error.
Twiss works with `q0=1` and `q0=2`, but not greater than 3. I see in the documentation, `q0` is in units of `[e]`, so presumably should be an integer charge.

Minimum working notebook is attached in zip, with two xsuite json sequences - thinline and thickline. Sequences available upon request.
[XSuiteIons.zip](https://github.com/xsuite/xsuite/files/12071235/XSuiteIons.zip)

---
Following code runs successfully:
```
import xtrack as xt
import xpart as xp
import xobjects as xo

thinline = xt.Line.from_json('PIMMS_thin.json')
thinline.build_tracker()

mass = 11.177929 * 1E9 # GeV
pc = 0.951303*12 * 1E9 # GeV
charge = 6.0

particles = xp.Particles(mass0=mass, q0=charge, p0c=pc, nparticles=100)
thinline.particle_ref = particles
thinline.twiss_default['method'] = '4d'
thintwiss = thinline.twiss()
```
Following code receives `Warning! Need second attempt on closed orbit search`
`ClosedOrbitSearchError`
```
thickline = xt.Line.from_json('PIMMS_thick.json')
thickline.configure_bend_model(core='full', edge='full')
thickline.build_tracker()

particles = xp.Particles(mass0=mass, q0=charge, p0c=pc, nparticles=100)
thickline.particle_ref = particles
thickline.twiss_default['method'] = '4d'
thicktwiss = thickline.twiss()
```
Following code uses default particle setup and is successful:
```
particles = xp.Particles(p0c=pc, nparticles=100)
thickline.particle_ref = particles
thickline.twiss_default['method'] = '4d'

thicktwiss = thickline.twiss()
```

**Comment by giadarol** [comment]
Thanks for reporting this, it is indeed a bug: in the dipole wedge the strength is erroneously multiplied by the charge.
A release with the corresponding fix will come very soon.

Indeed we had not tested the thick elements with ions (just added it to the todo list: #370)

**Comment by giadarol** [comment]
Solved in v0.38.3. Don't hesitate to reopen in case you still face issues.
